core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/scan/text/FileScanSourceOpDesc.scala

```diff
@@ -0,0 +1,38 @@
+package edu.uci.ics.texera.workflow.operators.source.scan.text
+
+import com.fasterxml.jackson.annotation.{JsonIgnoreProperties, JsonProperty}
+import com.kjetland.jackson.jsonSchema.annotations.{
+  JsonSchemaInject,
+  JsonSchemaString,
+  JsonSchemaTitle
+}
+import edu.uci.ics.amber.engine.architecture.deploysemantics.layer.OpExecConfig
+import edu.uci.ics.texera.workflow.common.metadata.annotations.HideAnnotation
+import edu.uci.ics.texera.workflow.common.tuple.schema.{Attribute, OperatorSchemaInfo, Schema}
+import edu.uci.ics.texera.workflow.operators.source.scan.{FileDecodingMethod, ScanSourceOpDesc}
+
+@JsonIgnoreProperties(value = Array("limit", "offset", "fileEncoding"))
+class FileScanSourceOpDesc extends ScanSourceOpDesc with TextSourceOpDesc {
+  @JsonProperty(defaultValue = "UTF_8", required = true)
+  @JsonSchemaTitle("Encoding")
+  @JsonSchemaInject(
+    strings = Array(
+      new JsonSchemaString(path = HideAnnotation.hideTarget, value = "attributeType"),
+      new JsonSchemaString(path = HideAnnotation.hideType, value = HideAnnotation.Type.equals),
+      new JsonSchemaString(path = HideAnnotation.hideExpectedValue, value = "binary")
+    )
+  )
+  var encoding: FileDecodingMethod = FileDecodingMethod.UTF_8
+
+  @JsonProperty(defaultValue = "false")
+  @JsonSchemaTitle("Unzip")
+  var unzip: Boolean = false
```

Comment:
There are many "zip" technologies, such as 7zip, gzip, tar, etc. Do we consider add them in the future? maybe we can generalize this operator by 
1. Make this check box as `unarchive` or `uncompress` instead of `unzip`;
2. let the user select the target unarchive (e.g., unzip, gunzip, tar) technology, or detect it from the file? There exist tools (e.g., unix command `file`) to determine if a file is compressed, and which format it is.

Reply:
Renamed to **Extract**.

Reply:
Replace zip with Apache Commons Compress™ which can take gzip, bzip2, XZ, LZMA, Pack200, DEFLATE, Brotli, DEFLATE64, ZStandard, Z, 7z, ar, arj, cpio, dump, tar, and zip